### PR TITLE
recent search history duplication fix

### PR DESCRIPTION
**Pull request by alias01**

### CODE REVIEW

File: src/store/map/index.js [hunk 1]
```diff
@@ -57,50 +57,54 @@ const getters = {
...
+        state.selectedLocation.locName = location.place_name;
 
-        const lsRecentID = JSON.parse(localStorage.getItem('recentID'));
-        if (lsRecentID !== undefined) {
-            state.recentID = lsRecentID;
-        }
-
-        const objData = {
-            id: state.recentID,
+        const newLocalRecentSearch = {
```
Comment (arrogant-ant) [inline src/store/map/index.js:68]: does local in the name mean variable local in the scope of method.
If so, then its misleading

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/components/extras/SearchInput.vue [hunk 2]
```diff
@@ -35,7 +35,7 @@
                         </div>
                     </div>
                 </template>
-                <template #empty>No results for {{ search }}</template>
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/extras/SearchInput.vue:38]: done

File: src/components/extras/SearchInput.vue [hunk 3]
```diff
@@ -86,9 +86,9 @@ export default {
         selected(newValue) {
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/extras/SearchInput.vue:86]: done

File: src/components/extras/SearchInput.vue [hunk 4]
```diff
@@ -121,9 +121,9 @@ export default {
         }, 500),
 
         async addRecentSearches() {
-            const recentLocations = await this.getFromRecent();
+            const recentSearch = await this.getFromRecent();
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/extras/SearchInput.vue:124]: done

File: src/components/extras/SearchInput.vue [hunk 5]
```diff
@@ -121,9 +121,9 @@ export default {
...
             if (this.LocationName.length === 0) {
-                for (const value of recentLocations) {
+                for (const value of recentSearch) {
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/extras/SearchInput.vue:126]: done

File: src/store/map/index.js (hunk 1 above)
Comment (alias01) [reply src/store/map/index.js:68]: yes the variable has the local scope. How so it is misleading

File: src/store/map/index.js [hunk 6]
```diff
@@ -57,50 +57,54 @@ const getters = {
 };
 
 const mutations = {
-    'update-location'(state, data) {
-        const newData = [...state.recentSearch, ...data];
-        state.locations = [...newData];
+    'update-location'(state, location) {
+        const localLocation = [...state.recentSearch, ...location];
+        state.location = [...localLocation];
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/store/map/index.js]: done

File: src/store/map/index.js (hunk 1 above)
Comment (alias01) [reply src/store/map/index.js]: done

Comment (arrogant-ant) [reply src/store/map/index.js]: please consider using _ instead local as its short and has no other meaning on contrary local has its own meaning too.
https://www.robinwieruch.de/javascript-naming-conventions/

File: src/store/map/index.js [hunk 7]
```diff
@@ -2,7 +2,7 @@ import { mapBoxClient, mayaClient } from '@/services/api';
 import _ from 'lodash';
 
 const state = {
-    locations: [],
+    location: [],
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/store/map/index.js]: done

**Review by arrogant-ant — APPROVED** [review]

**Pull request merged**